### DISCUSSION TITLE
How to load the converted .hic file into Juicerbox

### DISCUSSION THREAD

**Discussion by Nuturetree** [paulsengroup/hictk#192] [Q&A] [answered]
The hic file regenerated by the order：
hictk load genome.sizes A1.hic -f validpairs -b 40000 < A1.allValidPairs

However the .hic file can not load into the juicerbox, how to resolved the err.
[A1.zip](https://github.com/user-attachments/files/15972357/A1.zip)

**Comment by robomics** [comment, marked as answer]
Hi!

I can open the file just fine on my laptop:
![image](https://github.com/paulsengroup/hictk/assets/71787608/4f72cd06-7d22-41f2-9e6f-2a8e03790682)

The .hic files generated by hictk are in .hic v9 format (the latest revision).
Older versions of juicebox cannot open these files, and I suspect this may be the problem in your case.

Can you try to open the files using juiceboxGUI (instead of normal Juicebox): https://github.com/aidenlab/JuiceboxGUI/releases/tag/v3.1.4

According to Juicebox's [README](https://github.com/aidenlab/Juicebox/blob/982d1fa5e53148acc98ba3b56768dd952b3f5298/README.md) newer versions of Juicebox will be published in the JuiceboxGUI repository.

**Comment by Nuturetree** [comment]
thanks
I have a other question, how to convert the validpairs to the matrix (Bin1 bin2 reads).

**Comment by robomics** [comment]
Did the above suggestion solve your issue?

hictk cannot directly convert validpairs to the COO format you are referring to.
However, once you have the validpairs loaded into a .hic or .cool file, then you can use `hictk dump` to view interactions in COO format.

See the [docs](https://hictk.readthedocs.io/en/v1.0.0/reading_interactions.html) for an example of how this can be done (you probably need to run something like `hictk dump mymatrix.hic --resolution 10000`).

**Comment by Nuturetree** [comment]
Could this feature be added, as the matrix format generated by HiC-Pro is also important for downstream analysis?

**Comment by robomics** [comment]
Can you paste here the head of the matrix file you are talking about?

If I remember correctly HiC-Pro matrix files are in one-based COO format already.
If this is the case (and I understand your question correctly), then you can load the pre-binned interactions using `hictk load -f coo --one-based ... < file.matrix`

**Comment by Nuturetree** [comment]
thanks
